Utilities/bootstrap

```diff
@@ -285,11 +282,12 @@ def get_ninja_path(args):
         return call_output(["which", "ninja"], verbose=args.verbose)
 
 def get_build_target(args, cross_compile=False):
-    """Returns the target-triple of the current machine."""
+    """Returns the target-triple of the current machine or for cross-compilation."""
     try:
         command = [args.swiftc_path, '-print-target-info']
         if cross_compile:
-            command += ['-target', args.cross_compile_target_triple]
+            cross_compile_json = json.load(open(args.cross_compile_config))
+            command += ['-target', cross_compile_json["target"]]
```

Comment:
Alright, this works and allows me to remove the extra triple flag. I just realized that the destination.json is just a JSON file, so we can parse it to get the target triple.

@abertelrud, if you agree with this change, just let me know and I will squash it into the first commit and update the commit message and we can get this in.

Reply:
Thanks @buttaface, simplifying this down to get the information from the JSON instead of the having to pass another flag is a good improvement.  I still want to hear what @neonichu and @tomerd think but this looks good given how things currently work.  I will still want to work with @shahmishal to see if we can simplify the conceptual model around building for various platforms and architectures.

Reply:
Alright, will squash this commit and update the message.